File: solutions/go/darts/2/darts.go
// Package darts helps calculate scores for the game of darts.
package darts

import "math"

// Score returns the score earned by a dart throw.
func Score(x, y float64) int {
	switch dist := math.Hypot(x, y); {
	case dist <= 1:
		return 10
	case dist <= 5:
		return 5
	case dist <= 10:
		return 1
	default:
		return 0
	}
}

File: solutions/c/darts/2/darts.c
#include <inttypes.h>
#include <math.h>

#include "darts.h"

uint8_t score(coordinate_t landing_position) {
    float dist_to_radius = hypot(landing_position.x, landing_position.y);

    if (dist_to_radius <= 1) return 10;
    else if (dist_to_radius <= 5) return 5;
    else if (dist_to_radius <= 10) return 1;
    else return 0;
}